docs/docs/developers/tutorials/contract_tutorials/token_contract.md

```diff
@@ -0,0 +1,629 @@
+
+- How many BOB tokens they've actually used
+- When they're using mental health services
+- Their therapy patterns or frequency
+
+In this tutorial, we'll build a token contract that allows Giggle to mint BOB tokens for employees while ensuring complete privacy in how those tokens are spent.
+
+## Prerequisites
+
+This is an intermediate tutorial that assumes you have:
+- Completed the [Counter Contract tutorial](./counter_contract.md)
+- A running Aztec Sandbox (see the Counter tutorial for setup)
+- Basic understanding of Aztec.nr syntax and structure
+- Aztec toolchain installed (`aztec-up -v #include_version_without_prefix`)
+
+If you haven't completed the Counter Contract tutorial, please do so first as we'll skip the basic setup steps covered there.
+
+## What We're Building
+
+We'll create BOB tokens with:
+
+- **Public and Private minting**: Giggle can mint tokens in private or public
+- **Public and Private transfers**: Employees can spend tokens at Bob's clinic with full privacy
+
+### Project Setup
+
+Let's create a simple yarn + aztec.nr project:
+
+```bash
+yarn init
+yarn add @aztec/aztec.js@#include_aztec_version @aztec/accounts@#include_aztec_version
+aztec-nargo init --contract
+```
+
+## Contract structure
+
+We have a messy, but working structure. In `src/main.nr` we even have a proto-contract. Let's replace it with a simple starting point:
+
+```noir
+use aztec::macros::aztec;
+#[aztec]
+pub contract BobToken {
+  // We'll build the mental health token here
+}
+```
+
+The `#[aztec]` macro transforms our contract code to work with Aztec's privacy protocol. We'll rename it from `StarterToken` to `BobToken` to reflect our use case.
+
+Let's import the Aztec.nr library by adding it to our dependencies in `Nargo.toml`:
+
+```toml
+[package]
+name = "bob_token_contract"
+type = "contract"
+
+[dependencies]
+aztec = { git = "https://github.com/AztecProtocol/aztec-packages/", tag = "#include_aztec_version", directory = "noir-projects/aztec-nr/aztec" }
+```
+
+Since we're here, let's import more specific stuff from this library:
+
+```noir
+pub contract BobToken {
+    use aztec::{
+        macros::{functions::{initializer, private, public, utility, internal}, storage::storage},
+        protocol_types::{address::AztecAddress, traits::ToField},
+        state_vars::Map,
+        state_vars::public_mutable::PublicMutable,
+    };
+}
+```
+
+These are the different macros we need to define the visibility of functions, and some handy types and functions.
+
+## Building the Mental Health Token System
+
+### The Privacy Architecture
+
+Before we start coding, let's understand how privacy works in our mental health token system:
+
+1. **Public Layer**: Giggle mints tokens publicly - transparent and auditable
+2. **Private Layer**: Employees transfer and spend tokens privately - completely confidential
+3. **Cross-layer Bridge**: Employees can move tokens between public and private domains as needed
```

Comment:
```suggestion
3. **Cross-layer Transfer**: Employees can move tokens between public and private domains as needed
```
Bridge means something specific in regards to moving across chains so may be best to avoid the word